lib/src/channel_list_view.dart

```diff
@@ -259,22 +266,35 @@ class _ChannelListViewState extends State<ChannelListView>
           }
 
           if (channels.isNotEmpty) {
-            child = ListView.custom(
-              physics: AlwaysScrollableScrollPhysics(),
-              controller: _scrollController,
-              childrenDelegate: SliverChildBuilderDelegate(
-                (context, i) {
-                  return _itemBuilder(context, i, channels);
+            if (widget.crossAxisCount > 1) {
+              child = GridView.builder(
+                gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
+                    crossAxisCount: widget.crossAxisCount),
+                itemCount: channels.length,
+                physics: AlwaysScrollableScrollPhysics(),
+                controller: _scrollController,
+                itemBuilder: (context, index) {
+                  return _gridItemBuilder(context, index, channels);
                 },
-                childCount: (channels.length * 2) + 1,
-                findChildIndexCallback: (key) {
-                  final ValueKey<String> valueKey = key;
-                  final index = channels.indexWhere(
-                      (channel) => 'CHANNEL-${channel.id}' == valueKey.value);
-                  return index != -1 ? (index * 2) : null;
-                },
-              ),
-            );
+              );
+            } else {
+              child = ListView.custom(
+                physics: AlwaysScrollableScrollPhysics(),
+                controller: _scrollController,
+                childrenDelegate: SliverChildBuilderDelegate(
+                  (context, i) {
+                    return _itemBuilder(context, i, channels);
+                  },
+                  childCount: (channels.length * 2) + 1,
+                  findChildIndexCallback: (key) {
+                    final ValueKey<String> valueKey = key;
+                    final index = channels.indexWhere(
+                        (channel) => 'CHANNEL-${channel.id}' == valueKey.value);
+                    return index != -1 ? (index * 2) : null;
+                  },
+                ),
+              );
+            }
```

Comment:
You might want to change it to normal `ListView.builder` as it sometimes throws some index error. I did similar in `UserListView`

Reply:
That code was already there, did not add that.